lms/events/event.py

```diff
@@ -101,3 +102,33 @@ def _get_assignment_id(self):
             return assignment.id
 
         return None
+
+
+class AdminChangeEvent(BaseEvent):
+    def __init__(self, instance, *args, **kwargs):
+        self._instance = instance
+        super().__init__(*args, **kwargs)
+
+    def _get_data(self):
+        changes = {}
+        instance_details = inspect(self._instance)
+        for attr in instance_details.attrs:
+            history = instance_details.get_history(attr.key, True)
+
+            if not history.has_changes():
+                continue
+
+            changes[attr.key] = (1, history.added[0])
+
+        return {"user": self.request.identity.userid, "changes": changes}
+
+    @staticmethod
+    def notify_changes(request, instance):
+        if request.db.is_modified(instance):
+            request.registry.notify(
+                AdminChangeEvent(
+                    instance=instance,
+                    request=request,
+                    type=AdminChangeEvent.Type.ADMIN_PAGES_UPDATE,
```

Comment:
I think the type is implicit from raising an `AdminChangeEvent`? Or is the idea that this is like my idea for the separation between the audit event and the source?